needle-zendesk-connector/src/server/backend/connectors-backend.ts

```diff
@@ -0,0 +1,257 @@
+
+import { db } from "../db";
+import { filesTable, zendeskConnectorsTable } from "../db/schema";
+import { eq } from "drizzle-orm";
+import { createZendeskService } from "../zendesk/service";
+import {
+  getConnectorDetails,
+  getCurrentFiles,
+  handleDatabaseUpdates,
+} from "./zendesk-db";
+import { type DbFile } from "../zendesk/types";
+import { computeDiff } from "./diff-utils";
+
+export async function createZendeskConnector(
+  params: CreateConnectorRequest,
+  session: Session,
+) {
+  const connector = await createConnector(
+    {
+      name: params.name,
+      cronJob: "0 0 * * *",
+      cronJobTimezone: "Europe/Berlin",
+      collectionIds: [params.collectionId],
+      credentials: params.credentials,
+    },
+    session.id,
+  );
+
+  const descriptor: ConnectorRunDescriptor = {
+    create: [],
+    update: [],
+    delete: [],
+  };
+  const fileIdMap = new Map<string, string>();
+
+  // Process tickets
+  for (const ticket of params.selectedTickets) {
+    const ticketFileId = createNeedleFileId();
+    const commentsFileId = createNeedleFileId();
+
+    fileIdMap.set(`ticket-${ticket.id}`, ticketFileId);
+    fileIdMap.set(`comments-${ticket.id}`, commentsFileId);
+
+    descriptor.create.push(
+      {
+        id: ticketFileId,
+        url: ticket.url,
+        type: "text/plain",
+      },
+      {
+        id: commentsFileId,
+        url: ticket.url.replace(".json", "/comments.json"),
+        type: "text/plain",
+      },
+    );
+  }
+
+  // Process articles
+  for (const article of params.selectedArticles) {
+    const articleFileId = createNeedleFileId();
+    fileIdMap.set(`article-${article.id}`, articleFileId);
+
+    descriptor.create.push({
+      id: articleFileId,
+      url: article.html_url,
+      type: "text/html",
+    });
+  }
+
+  await db.transaction(async (tx) => {
+    await tx.insert(zendeskConnectorsTable).values({
+      connectorId: connector.id,
+      subdomain: params.subdomain,
+      orgId: params.organizationId,
+      includeTickets: params.selectedTickets.length > 0,
+      includeArticles: params.selectedArticles.length > 0,
+    });
+
+    await handleDatabaseUpdates(
+      tx,
+      connector.id,
+      { create: params.selectedTickets, update: [], delete: [] },
+      { create: params.selectedArticles, update: [], delete: [] },
+      fileIdMap,
+    );
+  });
+
+  await publishConnectorRun(connector.id, descriptor);
+  return connector;
+}
+
+export async function runZendeskConnector(
+  { connectorId }: { connectorId: string },
+  session?: Session,
+) {
+  let accessToken = "";
+  if (session) {
+    const { credentials } = await getConnector(connectorId, session.id);
+    accessToken = credentials ?? "";
+  }
+
+  const connectorDetails = await getConnectorDetails(connectorId);
+  if (!connectorDetails) {
+    throw new Error(`No Zendesk connector found for ID: ${connectorId}`);
+  }
+
+  const currentFiles = await getCurrentFiles(connectorId);
```

Comment:
Can we support attachments in the tickets as well?

Reply:
I see, this will introduce more complexity since attachments are kept as an array we can loop through and append them to save in the DB. Files can be many things (JPEG, PDF, TXT etc.) and file sizes can vary significantly. Should we implement this in a separate PR to better handle these considerations?